apps/platform-app/src/app/[locale]/(profile)/_workflows/profile-evaluation.workflow.ts

```diff
@@ -0,0 +1,152 @@
+  {
+    name: 'accept_profile',
+    description: 'Accept user apply and save his profile in our database',
+    schema: z.object({}),
+  },
+)
+
+const rejectProfileTool = tool(
+  async (input, config: LangGraphRunnableConfig) => {
+    const userId = config.configurable?.userId
+    const { reason } = input
+
+    const currentState =
+      getContextVariable<typeof StateAnnotation.State>('currentState')
+
+    if (currentState?.user) {
+      if (currentState.user.profile) {
+        updateProfileById(currentState.user.profile.id, { state: 'REJECTED' })
+      }
+      sendProfileRejectedEmail(currentState.user.email, reason)
+    }
+
+    return `The profile has been rejected. Reason: ${reason}, UserId: ${userId}`
+  },
+  {
+    name: 'reject_profile',
+    description: 'Reject user apply and informing him about reason',
+    schema: z.object({
+      reason: z
+        .string()
+        .describe('The reason why did you rejected user profile'),
+    }),
+  },
+)
+
+const tools = [acceptProfileTool, rejectProfileTool]
+
+import { evaluateProfilePrompt } from './prompts/evaluateProfileNode'
+import { executeDecisionPrompt } from './prompts/executeDecisionNode'
+async function retrieveProfile(
+  state: typeof StateAnnotation.State,
+  config: LangGraphRunnableConfig,
+) {
+  const userId = config.configurable?.userId
+  const fetchedUser = await getUserById(userId)
+
+  return { user: fetchedUser }
+}
+
+const evaluateProfileByModel = async (state: typeof StateAnnotation.State) => {
+  const { profile } = state.user
+  const fullName = profile?.fullName || ''
+  const bio = profile?.bio || ''
+
+  const chain = evaluateProfilePrompt.pipe(evaluationModel)
+  const responseMessage = await chain.invoke({ fullName, bio })
+
+
+  return { evaluation: responseMessage.content }
+}
+
+const executeDecision = async (state: typeof StateAnnotation.State) => {
+  const { evaluation } = state
+
+  const modelWithTools = executionModel.bindTools(tools)
+  const chain = executeDecisionPrompt.pipe(modelWithTools)
+  const responseMessage = await chain.invoke({ evaluation })
+
+  const messageWithSingleToolCall = new AIMessage({
+    content: responseMessage.content,
+    tool_calls: responseMessage.tool_calls?.[0]
+      ? [responseMessage.tool_calls?.[0]]
+      : undefined,
+  })
+
+  setContextVariable('currentState', state)
+  const toolNode = new ToolNode(tools)
+  await toolNode.invoke({ messages: [messageWithSingleToolCall] })
+}
+
+const workflow = new StateGraph(StateAnnotation)
+  .addNode('retrieveProfile', retrieveProfile)
+  .addNode('evaluateProfile', evaluateProfileByModel)
+  .addNode('executeDecision', executeDecision)
+  .addEdge(START, 'retrieveProfile')
+  .addEdge('retrieveProfile', 'evaluateProfile')
+  .addEdge('evaluateProfile', 'executeDecision')
+  .addEdge('executeDecision', END)
+
+const graph = workflow.compile()
+
+export const runEvaluateProfileAgent = async (userId: string) => {
+  'use server'
+  await graph.invoke('', {
```

Comment:
**nit:** let's return value from here which will be a whole state. we will be able to use it later